src/swift/imap/IMAPSearchOperation.swift

```diff
@@ -16,7 +17,9 @@ public class MCOIMAPSearchOperation : MCOIMAPBaseOperation {
 
 	deinit {
         self.operation.release()
+        lock.lock()
         completionBlock = nil
+        lock.unlock()
```

Comment:
You don't need locks in de-init.
If you are here, and you have the execution of other functions on another thread in some way. Then we are already in trouble. Locks don't help us : )

Reply:
I don't see a reason why are we niling the property here. We should not get into any function after deinit.

Reply:
Agree, we can remove nilling as well.